dora/core/common/src/main/java/alluxio/uri/UfsUrl.java

```diff
@@ -0,0 +1,283 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.uri;
+
+import alluxio.AlluxioURI;
+import alluxio.conf.Configuration;
+import alluxio.conf.PropertyKey;
+import alluxio.exception.InvalidPathException;
+import alluxio.grpc.UfsUrlMessage;
+import alluxio.util.io.PathUtils;
+
+import com.google.common.base.Preconditions;
+import org.apache.logging.log4j.util.Strings;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Objects;
+
+public class UfsUrl {
+
+  public static final String SCHEME_SEPERATOR = "://";
+  public static final String PATH_SEPERATOR = "/";
+  UfsUrlMessage mProto;
+
+  public UfsUrl(UfsUrlMessage proto) {
+    Preconditions.checkArgument(proto.getPathComponentsList().size() != 0,
+        "The proto.path is empty, please check the proto first");
+    mProto = proto;
+  }
+
+  public UfsUrl(String ufsPath) {
```

Comment:
i'm actually thinking if it makes sense for the `UfsUrl` to just store the original string too. WDYT @dbw9580 ?